src/utils/updateWiki.ts

```diff
@@ -0,0 +1,272 @@
+import { statSync, readdirSync } from 'fs';
+import { appendFile, writeFile, readdir, readFile } from 'fs/promises';
+import { logger } from '../logger/default';
+
+// File Paths
+const wikiPath = 'docs/COMMAND-WIKI.md';
+const commandsDir = 'src/commands';
+const commandDetailsDir = 'src/commandDetails';
+
+// Commands with subcommands, to be handled separately
+const retrieveCompoundCommands = (): string[] => {
+  const compoundCommands: string[] = [];
+
+  const filesAndDirs = readdirSync(commandsDir);
+  const directories = filesAndDirs.filter((file) =>
+    statSync(`${commandsDir}/${file}`).isDirectory(),
+  );
+
+  for (const dir of directories) {
+    const subDir = `${commandsDir}/${dir}`;
+    const files = readdirSync(subDir);
+    if (files.length === 1 && files[0] === `${dir}.ts`) {
+      compoundCommands.push(dir);
+    }
+  }
+
+  return compoundCommands;
+};
+
+// As of Feb 8 2024, this should be ['coin', 'company', 'interviewer', 'leetcode', 'profile'];
+const compoundCommands: string[] = retrieveCompoundCommands();
+
+// RegEx patterns to extract info
+const commandDetailsPattern = /.*const .+CommandDetails: CodeyCommandDetails = {([\s\S]*?)\n}/;
+const namePattern = /name: '(.*?)'/;
+const aliasesPattern = /aliases: \[([\s\S]*?)\]/;
+const descriptionPattern = /description: '(.*?)'/;
+const detailedDescriptionPattern = /detailedDescription: `([\s\S]*?)`,/;
+const optionsPattern = /options: \[([\s\S]*?)\]/;
+const subcommandDetailsPattern = /subcommandDetails: {([\s\S]*?)},/;
+
+// ----------------------------------- START OF UTILITY FUNCTIONS ---------------------------- //
+
+const formatCommandName = async (name: string | undefined) => {
+  let formattedName = '';
+  if (name === undefined) {
+    formattedName = 'None';
+  } else {
+    formattedName = name;
+  }
+  await appendFile(wikiPath, `## ${formattedName}\n`);
+};
+
+const formatSubcommandName = async (name: string | undefined, basecmd: string) => {
+  let formattedName = '';
+  if (name === undefined) {
+    formattedName = 'None';
+  } else {
+    formattedName = name;
+  }
+  await appendFile(wikiPath, `## ${basecmd} ${formattedName}\n`);
+};
```

Comment:
- Since the functionality doesn't seem to differ too much between Command and Subcommand, maybe we can just get rid of the Subcommand version and for subcommands just also pass in the base name with the subcommand name
- It would be odd to provide details/examples/options etc for a command without a name, so if the command name can't be extracted maybe log an error (maybe print `codeSection`) and just return (from the proposed `extractAndFormat` function below) without appending so the entire command won't appear in the wiki